src/test/java/com/jcabi/github/RtStarsITCase.java

```diff
@@ -0,0 +1,104 @@
+ * are met: 1) Redistributions of source code must retain the above
+ * copyright notice, this list of conditions and the following
+ * disclaimer. 2) Redistributions in binary form must reproduce the above
+ * copyright notice, this list of conditions and the following
+ * disclaimer in the documentation and/or other materials provided
+ * with the distribution. 3) Neither the name of the jcabi.com nor
+ * the names of its contributors may be used to endorse or promote
+ * products derived from this software without specific prior written
+ * permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT
+ * NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
+ * FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL
+ * THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,
+ * INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+ * STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED
+ * OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.jcabi.github;
+
+import java.io.IOException;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.Matchers;
+import org.junit.AfterClass;
+import org.junit.Assume;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+/**
+ * Integration test case for {@link RtStars}.
+ *
+ * @author Paul Polishchuk (ppol@ua.fm)
+ * @version $Id$
+ */
+public final class RtStarsITCase {
+    /**
+     * Test repos.
+     */
+    private static Repos repos;
+
+    /**
+     * Test repo.
+     */
+    private static Repo repo;
+
+    /**
+     * Set up tests.
+     * @throws IOException If some errors occurred.
+     */
+    @BeforeClass
+    public static void setUp() throws IOException  {
+        final String key = System.getProperty("failsafe.github.key");
+        Assume.assumeThat(key, Matchers.notNullValue());
+        final Github github = new RtGithub(key);
+        repos = github.repos();
+        repo = new RepoRule().repo(repos);
+    }
+
+    /**
+     * Set up tests.
+     * @throws IOException If some errors occurred.
+     */
+    @AfterClass
+    public static void tearDown() throws IOException  {
+        if (repos != null && repo != null) {
+            repos.remove(repo.coordinates());
+        }
+    }
+
+    /**
+     * Test star, unstar and check if starred functionalities.
+     * @throws IOException If some errors occurred.
+     */
+    @Test
+    public void testStarring() throws IOException {
+        final String repoName = repo.coordinates().repo();
+        final String userName = repo.coordinates().user();
```

Comment:
I can't use `repo` cos I already have class field with such name.

Regarding `coordinates` and `repo`, for me, when I read ` repo.stars().starred(user, coordinates)`, I wonder what is `coordinates`? `repoName` is much more clear. I don't need to refer the variable declaration to get the meaning of the variable. The meaning goes directly from variable name.

It's like with the method name `repo.coordinates().repo()`. What `repo()` returns ? I have to go to javadoc to figure out that it return the repository name. When we have `repo.coordinates().repoName()` I have this knowledge straight from the method name.

Of course, variable/method name should be concise, but it also should convey the meaning as clear as possible.

I'd love to hear your thought about this )  
(I do see the no-composite-variabe-name principle, but since I don't understand its profit, it's hard for me to follow it.)

For this question not to stop the work, I've renamed the variables to `coordinates`, `repo`.